### DISCUSSION TITLE
VRChat Hotkeys OSC

### DISCUSSION THREAD

**Discussion by ameliendeshams** [vrchat-community/osc#104] [Show and tell]
# VRChat Hotkeys OSC

A VRChat OSC software that allow you to **create Hotkeys and send OSC signals to VRChat** using your **keyboard** to control your avatar parameters.

[![Hotkeys OSC Demo](http://img.youtube.com/vi/2cf9lABm0m4/0.jpg)](https://youtu.be/2cf9lABm0m4)

## 👀 Nice, but what is the purpose?

It was first created for me, a desktop player :)

I use a **SoundPad** very often with predefined text to speech, I have more than 300+ differents sentences, on which I have assigned hotkeys to trigger some of them.

So... I have a lot of keyboard combinations to interact with other players,
it's very fun and often very interesting :)

But with the new OSC feature, I saw an opportunity to greatly improve my roleplay, by **controling my avatar animations with keyboard combinations** like **SoundPad**.

Now combined, I can trigger animations, toggle objects... **in the same time as I use my SoundPad**.

[![Hotkeys OSC Demo](http://img.youtube.com/vi/Ur3breZN2Pw/0.jpg)](https://youtu.be/Ur3breZN2Pw)

## 🌟 Features

<p align="center">
  <img src="https://gitlab.com/ameliend/vrchat-hotkeys-osc/-/raw/main/vrchat_hotkeys_osc/resources/screenshot.png">
</p>

* Automatic avatars parameters detection (depending of which avatar you choose in game, hit the refresh button)
* Set value (Boolean, Integer, or Float)
* Toggle value (Boolean, Integer, or Float)
* Add an optional delay (when set a value, or between a toggle)
* Open / Save your configuration.

**Comment by momo-the-monster** [comment]
Thanks for sharing this, @ameliendeshams! It's been added to the [list](https://github.com/vrchat-community/osc/wiki/Resources#controllers).
